cli/command/container/hijack.go

```diff
@@ -1,99 +1,180 @@
 package container
 
 import (
+	"fmt"
 	"io"
 	"runtime"
 	"sync"
 
 	"github.com/Sirupsen/logrus"
 	"github.com/docker/cli/cli/command"
 	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/pkg/ioutils"
 	"github.com/docker/docker/pkg/stdcopy"
+	"github.com/docker/docker/pkg/term"
 	"golang.org/x/net/context"
 )
 
-// holdHijackedConnection handles copying input to and output from streams to the
-// connection
-// nolint: gocyclo
-func holdHijackedConnection(ctx context.Context, streams command.Streams, tty bool, inputStream io.ReadCloser, outputStream, errorStream io.Writer, resp types.HijackedResponse) error {
-	var (
-		err         error
-		restoreOnce sync.Once
-	)
-	if inputStream != nil && tty {
-		if err := setRawTerminal(streams); err != nil {
-			return err
-		}
-		defer func() {
-			restoreOnce.Do(func() {
-				restoreTerminal(streams, inputStream)
-			})
-		}()
+// The default escape key sequence: ctrl-p, ctrl-q
+var defaultEscapeKeys = []byte{16, 17}
```

Comment:
The best place to put it would probably be [here](https://github.com/moby/moby/blob/3d630498c3d7c8d23889df3cf006128a1d5ddad4/pkg/term/proxy.go#L18) so it could live next to the Escape proxy implementation that uses it. Doing that would require yet another change to `github.com/moby/moby` though.